cs/ccxt/ws/Client.cs

```diff
@@ -240,13 +244,30 @@ public void Connect()
                 }
                 catch (Exception ex)
                 {
+                    _connectSemaphore.Release();
```

Comment:
should we only release on a crash? what if we succeed?

Reply:
ouch,  where did I put that, what a stupid mistake